src/main/java/seedu/duke/data/notebook/Notebook.java

```diff
@@ -34,10 +40,140 @@ public Notebook(ArrayList<Note> notes) {
         return this.notes;
     }
 
+    public boolean checkPinned() {
+        for (Note note : notes) {
+            if (note.getPinned()) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public ArrayList<Note> getPinnedNotes() {
+        ArrayList<Note> noteList = new ArrayList<>();
+        for (int i = 0; i < notes.size(); i++) {
+            if (notes.get(i).getPinned()) {
+                noteList.add(notes.get(i));
+            }
+        }
+        return noteList;
+    }
+
+    public ArrayList<Note> getPinnedNotes(ArrayList<Note> noteArrayList) {
+        ArrayList<Note> noteList = new ArrayList<>();
+        for (int i = 0; i < noteArrayList.size(); i++) {
+            if (noteArrayList.get(i).getPinned()) {
+                noteList.add(noteArrayList.get(i));
+            }
+        }
+        return noteList;
+    }
+
+    public ArrayList<Note> getUnpinnedNotes() {
+        ArrayList<Note> noteList = new ArrayList<>();
+        for (int i = 0; i < notes.size(); i++) {
+            if (!notes.get(i).getPinned()) {
+                noteList.add(notes.get(i));
+            }
+        }
+        return noteList;
+    }
+
+    public ArrayList<Note> getUnpinnedNotes(ArrayList<Note> noteArrayList) {
+        ArrayList<Note> noteList = new ArrayList<>();
+        for (int i = 0; i < noteArrayList.size(); i++) {
+            if (!noteArrayList.get(i).getPinned()) {
+                noteList.add(noteArrayList.get(i));
+            }
+        }
+        return noteList;
+    }
+
+    /**
+     * Method compiles the ArrayList items and appends the items to a String.
+     * The ArrayList has already been sorted
+     * Method returns either top to bottom or bottom to top to account for ascending/descending sorting
+     *
+     * @return noteString String containing the notes sorted either ascending ot descending
+     */
+    public ArrayList<Note> getSortedList(Boolean isAscendingOrder, Boolean notebookToggle) {
```

Comment:
The Boolean notebookToggle can be renamed to isPinned or something similar to better identify its purpose.